h2o-core/src/main/java/water/fvec/FileVec.java

```diff
@@ -194,7 +194,7 @@ public static int calcOptimalChunkSize(long totalSize, int numCols, long maxLine
     }
     else {
       // New Heuristic
-      int minNumberRows = 10; // need at least 10 rows (lines) per chunk (core)
+      int minNumberRows = 200; // need at least 200 rows (lines) per chunk (core) - that's needed for files with headers
```

Comment:
Why you cannot fetch the next chunk in parser/guesser to read header which is spread over multiple chunk.